src/app/modules/item/components/user-progress-details/user-progress-details.component.ts

```diff
@@ -0,0 +1,95 @@
+import { AfterViewInit, OnDestroy, SimpleChanges } from '@angular/core';
+import { Component, EventEmitter, Input, Output, ViewChild, OnChanges } from '@angular/core';
+import { OverlayPanel } from 'primeng/overlaypanel';
+import { animationFrames, merge, ReplaySubject, Subscription } from 'rxjs';
+import { filter, mapTo, switchMap, take } from 'rxjs/operators';
+import { TeamUserProgress } from 'src/app/shared/http-services/get-group-progress.service';
+
+@Component({
+  selector: 'alg-user-progress-details',
+  templateUrl: './user-progress-details.component.html',
+  styleUrls: [ './user-progress-details.component.scss' ],
+})
+export class UserProgressDetailsComponent implements OnChanges, OnDestroy, AfterViewInit {
+
+  @Input() progress?: TeamUserProgress;
+  @Input() canAccess?: boolean;
+  @Input() target?: Element;
+
+  @Output() accessPermissions = new EventEmitter<void>();
+  @Output() hide = new EventEmitter<void>();
+
+  @ViewChild('panel') panel?: OverlayPanel;
+
+  // NOTE: when the progress changes, there is a time between closing the previous overlay and opening a new one
+  // `displayedProgress` is the progress displayed in the overlay
+  // This value is populated *only* when the overlay is shown, not before.
+  displayedProgress?: TeamUserProgress;
+  isSuccess?: boolean;
+  hasScore?: boolean;
+  forwardHideEvent = false;
+
+  private subscriptions = new Subscription();
+  private overlay$ = new ReplaySubject<{ previousTarget?: Element, nextTarget?: Element, progress?: TeamUserProgress }>(1);
+
+  private overlayWithProgress = this.overlay$.pipe(
+    filter(({ progress }) => !!progress && (progress.validated || progress.score > 0 || progress.timeSpent > 0)),
+  );
+  private open$ = this.overlayWithProgress.pipe(filter(({ previousTarget, nextTarget }) => !previousTarget && !!nextTarget));
+  private reopen$ = this.overlayWithProgress.pipe(
+    filter(({ previousTarget, nextTarget }) => !!nextTarget && !!previousTarget && previousTarget !== nextTarget),
+  );
+  private hide$ = merge(
+    this.overlay$.pipe(filter(({ nextTarget }) => !nextTarget)),
+    this.reopen$,
+  );
+  private openOrReopen$ = merge(
+    this.open$,
+    this.reopen$.pipe(
+      switchMap(data => {
+        if (!this.panel) throw new Error('panel must be defined');
+        return this.panel.onHide.asObservable().pipe(mapTo(data), take(1));
+      }),
+    )
+  );
```

Comment:
After reading everything, actually I am not convinced of the benefit of using subjects for such a case instead of simple if/else in onChange. There is no actual steam here nor combination with async requests (edit: ah maybe for combining on line 65?), it is just doing hide/show immediately when the input change. The overall conditional process is not very readable.